modules/apps/asset/asset-taglib/src/main/resources/META-INF/resources/asset_categories_navigation/js/AssetCategoriesNavigationTreeView.js

```diff
@@ -15,13 +15,44 @@
 import {Treeview} from 'frontend-js-components-web';
 import React from 'react';
 
-const AssetCategoriesNavigationTreeView = ({categories, namespace}) => {
+function buildNodes(vocabularies, categories) {
```

Comment:
That's the part I think can be improved (besides the Java stuff in which I'm basically doing what I think is ok, but :shrug: )